experimental/trusted_shuffler/server/src/http.rs

```diff
@@ -51,9 +51,10 @@ impl RequestHandler for HttpRequestHandler {
         // let new_uri = Uri::from_parts(parts).expect("Failed to create new URI");
         let path = request.uri().path();
         // And extend the URL to the backend.
-        let new_uri = format!("{}{}", self.backend_url, path)
-            .parse::<Uri>()
-            .expect("Couldn't parse URI for backend.");
+        let new_uri = match format!("{}{}", self.backend_url, path).parse::<Uri>() {
+            Ok(new_uri) => new_uri,
+            Err(error) => return Err(anyhow!("Couldn't parse URI for backend: {:?}", error)),
+        };
```

Comment:
I think this can be accomplished more idiomatically with map_err and the `?` operator.

Reply:
I didn't manage because of the return.

Reply:
Is that because there are additional type conversions going on that are not performed by `Into`? Otherwise normally it should work for early returns

Reply:
Fixed now.